src/errors.tsx

```diff
@@ -1,12 +1,14 @@
+import { GenericError } from "@auth0/auth0-spa-js";
+
 /**
  * An OAuth2 error will come from the authorization server and will have at least an `error` property which will
  * be the error code. And possibly an `error_description` property
  *
  * See: https://openid.net/specs/openid-connect-core-1_0.html#rfc.section.3.1.2.6
  */
-export class OAuthError extends Error {
-  constructor(public error: string, public error_description?: string) {
-    super(error_description || error);
+export class OAuthError extends GenericError {
+  constructor(public error: string, public error_description: string = '') {
```

Comment:
error_description can not be undefined in GenericError, so I modified this to match.

Reply:
```suggestion
  constructor(error: string, error_description?) {
```

Since `GenericError` error defines these as public, you probably don't need to mark them as public here (then you don't need to default `error_description` to an empty string)

Reply:
This might have caused a bug? Not sure if the `public` keywords were important. Bug report: https://github.com/auth0/auth0-react/issues/645